### ISSUE TITLE
Resizing terminal window to 2 lines makes xvi dump core

### ISSUE THREAD

**Issue by martinwguy** [martinwguy/xvi#42]
With xvi running in a terminal emulator under X, you can resize the window and xvi adjusts but if the window is resized to a height of two lines, xvi gets a segmentation fault in
```move_cursor_to_window()``` at the first instruction
```
move_cursor_to_window(win)
Xviwin          *win;
{
    Posn        *cp;

    cp = win->w_cursor;  // <-- Here
```
called from ```xvi_handle_event()```

If you start xvi up in a window that's already two lines high, it works fine (well, until you make the window taller then reduce it to two lines again).

Tested-on: Debian i386 "jessie"

**Comment by jvvv** [comment]
According to my backtrace, it looks like something is zeroing curwin... I notice that the argument to move_cursor_to_window in the backtrace is 0x0. Stepping through the resize event, curwin gets zeroed at window.c line 523, xvAdjustWindows(). The following patch should prevent the segfault:
--- a/src/windows.c
+++ b/src/windows.c
@@ -519,7 +519,7 @@ bool_t	do_clear;
 		to_go -= w->w_nrows;
 		spare -= (w->w_nrows - Pn(P_minrows));
 	    	w->w_nrows = 0;
-		if (curwin == w) {
+		if (curwin == w && w->w_last) {
 		    curwin = w->w_last;
 		}
 	    } else {

**Comment by martinwguy** [comment]
Fixed by 74b7bc99
